### ISSUE TITLE
Bug: (eslintignore breaks when cwd is using forward slash on windows)

### ISSUE THREAD

**Issue by noe132** [eslint/eslint#17042]
### Environment

Node version: 19.7.0
npm version: 9.5.0
Local ESLint version: 8.37.0
Global ESLint version: N/A
Operating System: windows 10 x64

### What parser are you using?

Default (Espree)

### What did you do?

When passing cwd option to eslint with a path with forward slash on windows (`C:/path/to/project`), eslintignore breaks and errors were reported from ignored files.

folder structure:
```
src/1.js
src/dist/2.js
.eslintignore
.eslintrc.js
lint.js
```
<details>
<summary>src/1.js</summary>

```js
console.log(1)
```
</details>

<details>
<summary>src/dist/2.js</summary>

```js
console.log(2)
```
</details>

<details>
<summary>.eslintignore</summary>

```
src/dist
```
</details>

<details>
<summary>lint.js</summary>

```js
const eslint = require('eslint');

const run = async (cwd) => {
  // 1. Create an instance.
  const instance = new eslint.ESLint({ cwd, extensions: ['.js'] });

  // 2. Lint files.
  const results = await instance.lintFiles(["./src"]);

  // 3. Format the results.
  const formatter = await instance.loadFormatter("stylish");
  const resultText = formatter.format(results);

  // 4. Output it.
  console.log(resultText);
};

const main = async () => {
  await run(__dirname);
  console.log('-----------');
  await run(__dirname.replaceAll("\\", "/"));
};

main();
```
</details>

<details>
<summary>.eslintrc.js</summary>

```
module.exports = {
  parserOptions: {
    ecmaVersion: 2020,
  },
  'root': true,
  'rules': {
    'semi': 'error',
  },
};
```
</details>

<details>
<summary>result</summary>

```
C:\src\eslinttest\src\1.js
  1:15  error  Missing semicolon  semi

✖ 1 problem (1 error, 0 warnings)
  1 error and 0 warnings potentially fixable with the `--fix` option.

-----------

C:\src\eslinttest\src\1.js
  1:15  error  Missing semicolon  semi

C:\src\eslinttest\src\dist\2.js
  1:15  error  Missing semicolon  semi

✖ 2 problems (2 errors, 0 warnings)
  2 errors and 0 warnings potentially fixable with the `--fix` option.
```
</details>

### What did you expect to happen?

eslintignore working as expected

### What actually happened?

linting ignored files

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

Where I get this problem? [Here](https://github.com/fi3ework/vite-plugin-checker/blob/20b739943495130c8119a4d255104704d0770dda/packages/vite-plugin-checker/src/checkers/eslint/main.ts#L56).
I'm using vite-plugin-checker on my project. It pass a `root` param recieved from `vite` to eslint `cwd` option which happens to be a path with forward slash on windows.

**Comment by nzakas** [comment]
That seems reasonable, though I'm curious if this is also an issue for `FlatESLint`?

**Comment by nzakas** [comment]
@mdjermanovic following up here, do you think this is also an issue for `FlatESLint`?

**Comment by fasttime** [comment]
`FlatESLint` does not care about `.eslintignore` files, but it seems to me that the supported ways to ignore stuff with the flat config all behave well on Windows even when `cwd` contains forward slashes.
* with `ignorePatterns: ['src/dist']` in the `FlatESLint` constructor options.
* with a global `ignores` pattern in the config: `{ ignores: ['src/dist'] }`.
* with a non-global `ignores` pattern like `ignores: ['src/dist/**']`.
* with a negated `files` pattern like `files: ['!src/dist/**']`.

Maybe there is more to test, but so far I could not reproduce the issue described here with the flat config.

**Comment by mdjermanovic** [comment]
I didn't manage to find any observable problems either, but it might be good to do `path.normalize()` anyway so that rules can always expect `context.cwd` with local path separators.

**Comment by snitin315** [comment]
Marking this as accepted :)

**Comment by mdjermanovic** [comment]
I can work on this.
